### PR TITLE
Handle anchor link attributes for redirects

### PR DESCRIPTION
**Pull request by alabiaga**
- if "data-saferedirectreason" attr is present in link, cancel click and post message to viewer about unsafe link.
- if "data-saferedirecturl" attr is present in link, modify link href with that attribute's value safe link.

b/117824669

### CODE REVIEW

**Comment by alabiaga** [comment]
cc/ @zhangsu

**Comment by alabiaga** [comment]
@choumx good point. Is this unlikely to happen given we install the navigation service first https://github.com/ampproject/amphtml/blob/master/src/amp.js#L103 and the amp doc's visibility is queued in the end, https://github.com/ampproject/amphtml/blob/master/src/amp.js#L123 ?  Or is my understanding of how that code works incorrect? Thanks for pointing this out.

**Comment by choumx** [comment]
You can try it out by simulating a very slow network in Chrome devtools.

**Comment by zhangsu** [comment]
@alabiaga Aaron, what's the result of the simulation that Will mentioned?

**Comment by alabiaga** [comment]
In that simulation, the amp doc is initially hidden but is transitioned to
being visible by the css boiler plate included and required in all Amp
docs. When blocking the v0.js request via chrome Dev tools, the doc and
link for our use case will be visible but the click handler has yet to be
registered on the element until the v0.js binary downloads.

**Comment by zhangsu** [comment]
For the email spec, isn't the body initially hidden by the CSS boilerplate?

```html
<style amp4email-boilerplate>body{visibility:hidden}</style>
```

File: src/service/navigation.js [hunk 1]
```diff
@@ -435,6 +440,36 @@ export class Navigation {
...
+   * Handles clicking on links with possible redirects.
+   * @param {!Event} e
+   * @param {!Element} target
+   * @return {boolean}
+   * @private
+   */
+  handleRedirect_(e, target) {
+    const safeRedirectAttribute = 'data-saferedirectreason';
+    const safeRedirectLink = target.getAttribute('data-saferedirecturl');
+    const isUnsafeRedirect = target.hasAttribute(safeRedirectAttribute);
```
Comment (jridgewell) [inline src/service/navigation.js]: I don't understand the naming of these two attributes.

**Comment by choumx** [comment]
Thanks for looking into that.  I'd prefer if we use a safer default that doesn't rely on implicit order of unrelated code.

Also, we should follow existing convention for `data-*` attributes, e.g. `data-amp-safe-redirect-url` and `data-amp-safe-redirect-reason`.

File: src/service/navigation.js (hunk 1 above)
Comment (choumx) [reply src/service/navigation.js]: `data-saferedirectreason` seems to imply that the link was deemed unsafe. Confusing name though.

**Comment by alabiaga** [comment]
Closing PR as this will live in an integration script.

**Pull request closed without merging**